runtime/gc_glue_java/MetronomeDelegate.cpp

```diff
@@ -1174,14 +1182,22 @@ MM_MetronomeDelegate::releaseExclusiveVMAccess(MM_EnvironmentBase *env, bool rel
 
 	--(mainGCThread->omrVMThread->exclusiveCount);
 	if (releaseRequired) {
-		_javaVM->internalVMFunctions->releaseExclusiveVMAccessMetronome(mainGCThread);
-		/* Set the exclusive access response counts to an unusual value,
-		 * just for debug purposes, so we can detect scenarios, when main
-		 * thread is waiting for Xaccess with noone requesting it before.
-		 */
-		_vmResponsesRequiredForExclusiveVMAccess = 0x7fffffff;
-		_jniResponsesRequiredForExclusiveVMAccess = 0x7fffffff;
+		--(mainGCThread->omrVMThread->exclusiveCount);
```

Comment:
did you mean to remove the line from before the 'if' check?

Reply:
I'm not sure why there's a double decrement in this case, but as you say, it's just inlining the other function and excluding the actual exclusive release if exclusive was held in the caller.

Reply:
I also think the use of the NoMutex version of the release VM access call is likely incorrect, but it's what the old code was doing.